File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        vector<int> v;
    if(x<0){
        return false;
    }
    if(x==0){
        return true;
    }
    if(x>0){
    while(x>0){
    int r=x%10;
    x=x/10;
    v.push_back(r);
    }}
    int left=0;
    int right=v.size()-1;
    while(left<=right){
    if(v[left]!=v[right]){
      return false;
    }
    else{
    left++;
    right--;
    }}
    return true;
    
    }
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        num=0
        y=x
        if(x<0):
            return False
        if(x==0):
            return True
        while(x>0):
            r=x%10
            num=(num*10)+r
            x=x//10
        if(y==num):
            return True
        else:
            return False